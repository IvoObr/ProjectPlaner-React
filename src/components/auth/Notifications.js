import React from 'react';

const Notifications = (props) => {
    const doShow = props.doShow;
    const message = props.message;
    return (
        (doShow) ?
            <div className=' absolute hideElement center-align'>
                <div className='left-float '>
                    <i className="white-text material-icons line-height-left large">error_outline</i>
                </div>
                <div className='right-float'>
                    <p className='line-height-right'>{message}</p>
                </div>
            </div>
        : null
    )
};

export default Notifications;