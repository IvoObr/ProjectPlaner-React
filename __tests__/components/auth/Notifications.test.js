import React from 'react';
import ReactDOM from 'react-dom';
import Notifications from '../../../src/components/auth/Notifications';
import {configure, shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({adapter: new Adapter()});

describe('<Notifications/>', () => {
    let wrapper = null;
    let props = null;

    beforeEach(() => {
        props = {
            doShow: true,
            message: "testing with jest"
        };

        wrapper = shallow(<Notifications {...props}/>);
    });

    it('should render Notifications', () => {
        expect(wrapper.find('div').length).toBe(3);
        expect(wrapper.find('p').text()).toBe(props.message);
    });

    it('should not render Notifications', () => {
        wrapper.setProps({ doShow: false });
        expect(wrapper.find('div').length).toBe(0);
    });

    it('matches snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
