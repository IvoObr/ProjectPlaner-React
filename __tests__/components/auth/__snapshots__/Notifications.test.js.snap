// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notifications/> matches snapshot 1`] = `
<div
  className=" absolute hideElement center-align"
>
  <div
    className="left-float "
  >
    <i
      className="white-text material-icons line-height-left large"
    >
      error_outline
    </i>
  </div>
  <div
    className="right-float"
  >
    <p
      className="line-height-right"
    >
      testing with jest
    </p>
  </div>
</div>
`;
